src/qt/coincontroldialog.cpp

```diff
@@ -449,48 +449,52 @@ void CoinControlDialog::updateLabels(WalletModel *model, QDialog* dialog)
     bool fWitness               = false;
 
     std::vector<COutPoint> vCoinControl;
-    std::vector<COutput>   vOutputs;
-    coinControl->ListSelected(vCoinControl);
-    model->getOutputs(vCoinControl, vOutputs);
-
-    BOOST_FOREACH(const COutput& out, vOutputs) {
-        // unselect already spent, very unlikely scenario, this could happen
-        // when selected are spent elsewhere, like rpc or another computer
-        uint256 txhash = out.tx->GetHash();
-        COutPoint outpt(txhash, out.i);
-        if (model->isSpent(outpt))
-        {
-            coinControl->UnSelect(outpt);
-            continue;
-        }
 
-        // Quantity
-        nQuantity++;
+    {
+        LOCK2(cs_main, model->wallet->cs_wallet);
```

Comment:
How can we avoid this really undesirable layer violation (`model->wallet->cs_wallet`)? This seems to go in the wrong direction with @ryanofsky approach to form clean layers for later process separation.